js/main.js

```diff
@@ -0,0 +1,144 @@
+
+const canvas = document.getElementById('wordSearchCanvas');
+const context = canvas.getContext('2d');
+
+
+function redraw() {
+  let height = canvas.getBoundingClientRect().height;
+  let width = canvas.getBoundingClientRect().width;
+
+  // console.log('Height is: ' + height + ' Width is: ' + width);
+}
+
+/**
+ * Returns a word seach game board that is created based of the dimensions
+ * of the allocated screen space with each letter being allocated it's
+ * own square within the board.
+ * @param {number} canvasHeight
+ * @param {number} canvasWidth
+ * @returns {Object}
+ */
+function createGameBoard(canvasHeight, canvasWidth) {
+  let gameBoard =
+      {startPointX: 0, startPointY: 0, gameHeight: 10, gameWidth: 10, board: []}
+
+  let heightScaler = canvasHeight / gameBoard.gameHeight;
+  let widthScaler = canvasWidth / gameBoard.gameWidth;
+
+  if (widthScaler > heightScaler) {
+    widthScaler = heightScaler;
+  } else {
+    heightScaler = widthScaler;
+  }
+
+  let middlePoint = canvasWidth / 2 - (widthScaler * gameBoard.gameWidth / 2);
+  gameBoard.startPointX = middlePoint;
+
+  currX = gameBoard.startPointX;
+  currY = gameBoard.startPointY;
+
+  for (let i = 0; i < gameBoard.gameHeight; ++i) {
+    let rowOfSquares = [];
+    for (let j = 0; j < gameBoard.gameWidth; ++j) {
+      let square = {
+        xCoord: currX,
+        yCoord: currY,
+        width: widthScaler,
```

Comment:
Currently each square has its own width field. Can different squares have different width/heights? If no - then squareWidth/height should be a field of GameBoard.

Reply:
Good point - Updated